styx-service-common/src/test/java/com/spotify/styx/storage/DatastoreStorageTest.java

```diff
@@ -809,82 +809,10 @@ public void runInTransactionShouldCallFunctionAndRollbackOnFailure() throws Exce
     }
 
     verify(transactionFunction).apply(storageTransaction);
-    verify(storageTransaction, never()).commit();
-    verify(storageTransaction).rollback();
   }
 
   @Test
-  public void runInTransactionShouldCallFunctionAndRollbackOnPreCommitConflict() throws Exception {
```

Comment:
Make sure the rollback/commit thing are tested by some other test cases, if not let's add some.